pkg/asset/machines/vsphere/machines.go

```diff
@@ -88,6 +105,26 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 	return machines, nil
 }
 
+// applyNetworkConfig this function will apply the static ip configuration to the networkDevice
+// field in the provider spec.  The function will use the desired zone to determine which config
+// to apply and then remove that host config from the hosts array.
+func applyNetworkConfig(hosts []*vsphere.Host, desiredZone string, provider *machineapi.VSphereMachineProviderSpec) []*vsphere.Host {
+	for index, host := range hosts {
+		if host.FailureDomain == "" || host.FailureDomain == desiredZone {
+			networkDevice := host.NetworkDevice
+			if networkDevice != nil {
+				provider.Network.Devices[0].IPAddrs = networkDevice.IPAddrs
+				provider.Network.Devices[0].Nameservers = networkDevice.Nameservers
+				provider.Network.Devices[0].Gateway4 = networkDevice.Gateway4
+				provider.Network.Devices[0].Gateway6 = networkDevice.Gateway6
+			}
+			hosts = append(hosts[:index], hosts[index+1:]...)
+			break
+		}
+	}
+	return hosts
```

Comment:
Should we be returning an error if there isn't an IP address assigned?  I ran in to a situation where a given zone didn't have an associated host and no address was assigned.  Another way we could handle this would be an update to the platform validation to catch this situation before the installer even gets to the machine asset generation.

Reply:
Ideally this type of check should be done during validation in my opinion.  I can update SPLAT-828 to include that validation step.  Looking at SPLAT-828, we do check that network.IPAddrs != 0 for static IP assignment.  This may be a defect related to dhcp assignment of IPs.  Would OCPBUGS-5201 catch this?